### PR TITLE
EPMRPP-35259 || Finish parent dao

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/dao/TestItemRepository.java [hunk 1]
```diff
@@ -33,6 +39,62 @@
...
+	 * Retrieve the {@link List} of the {@link TestItem#itemId} by launch ID, {@link StatusEnum#name()} and {@link TestItem#hasChildren} == false
+	 *
+	 * @param launchId {@link com.epam.ta.reportportal.entity.launch.Launch#id}
+	 * @param status   {@link StatusEnum#name()}
+	 * @return the {@link List} of the {@link TestItem#itemId}
+	 */
+	@QueryHints(value = @QueryHint(name = HINT_FETCH_SIZE, value = "1"))
+	@Query(value = "SELECT test_item.item_id FROM test_item JOIN test_item_results result on test_item.item_id = result.result_id "
+			+ " WHERE test_item.launch_id = :launchId AND NOT test_item.has_children AND result.status = cast(:status as status_enum)", nativeQuery = true)
+	Stream<BigInteger> streamIdsByNotHasChildrenAndLaunchIdAndStatus(@Param("launchId") Long launchId, @Param("status") String status);
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/ta/reportportal/dao/TestItemRepository.java]: Why BigInteger?

Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/dao/TestItemRepository.java]: In our DB ids are BIGINT, that's why their representation in code is also a BigInteger, and also it's a native query, it means that BIGINT won't be casted to Long implicitly, so the only way is to get it as BigInteger and then call 'longValue()' method.

File: src/main/java/com/epam/ta/reportportal/dao/LaunchRepository.java [hunk 2]
```diff
@@ -46,7 +46,8 @@
 	Stream<Long> streamIdsModifiedBefore(@Param("projectId") Long projectId, @Param("before") LocalDateTime before);
 
 	@Query(value = "SELECT l.id FROM Launch l WHERE l.status = :status AND l.projectId = :projectId AND l.lastModified < :before")
-	Stream<Long> streamIdsWithStatusModifiedBefore(@Param("projectId") Long projectId, @Param("status") StatusEnum status, @Param("before") LocalDateTime before);
+	Stream<Long> streamIdsWithStatusModifiedBefore(@Param("projectId") Long projectId, @Param("status") StatusEnum status,
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/ta/reportportal/dao/LaunchRepository.java]: So why here we user Long and it works, but in test item repo BigInteger?

Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/dao/LaunchRepository.java]: It's a JPQL query, so it casts BigInteger to Long implicitly

**Pull request merged**